cicecore/shared/ice_calendar.F90

```diff
@@ -134,99 +164,172 @@ subroutine init_calendar
       if (icepack_warnings_aborted()) call abort_ice(error_message=subname, &
          file=__FILE__, line=__LINE__)
 
+      seconds_per_day = nint(secday)
+      if ((abs(real(seconds_per_day,kind=dbl_kind)/secday)-1.0_dbl_kind) > 1.0e-7) then
+         write(nu_diag,*) trim(subname),' ERROR secday should basically be an integer',secday
+         call abort_ice(subname//'ERROR: improper secday')
+      endif
+
       istep = 0         ! local timestep number
-      time=istep0*dt    ! s
-      yday=c0           ! absolute day number
-      mday=0            ! day of the month
-      month=0           ! month
-      nyr=0             ! year
-      idate=00000101    ! date
-      sec=0             ! seconds into date
+      myear=year_init   ! year
+      mmonth=month_init ! month
+      mday=day_init     ! day of the month
+      msec=sec_init     ! seconds into date
       istep1 = istep0   ! number of steps at current timestep
                         ! real (dumped) or imagined (use to set calendar)
```

Comment:
likewise, this comment could be removed to avoid confusion
~~~fortran
                        ! real (dumped) or imagined (use to set calendar)
~~~

Reply:
The istep0 question is a reasonable one.  I think I left it in there so a user could define the initial step number as something other than 0 if they wanted.  The key is that the steps still need to be counted and tracked on restart and they are.  I think if we remove istep0 from namelist and it is effectively hardwired to 0, then we can probably get rid of it in the entire code.  I think we should discuss further.